### PR TITLE
Disable more compression methods for minizip

### PR DESCRIPTION
**Pull request by Jan200101**
Doing this because I ran into the following error while compiling with msvc-wine
```
z:/home/sentry/git/NorthstarLauncher/thirdparty/minizip/mz_strm_zstd.c(17): fatal error C1083: Cannot open include file: 'zstd.h': No such file or directory
```

What is actually needed from minizip?
This was likely not an issue on windows because of `MZ_FETCH_LIBS`. Should we turn that off to ensure no extra dependencies get in without our knowledge?

### CODE REVIEW

**Comment by Jan200101** [comment]
> My guess is that `MZ_ZLIB` is required, but I have to try compiling without it to check.

ZLIB would be required for DEFLATE compressed zips (which is the most common format) but others are also possible.

Considering can control verified mods, should there be some kind of requirement what compression it needs to use?

File: cmake/Findminizip.cmake [hunk 1]
```diff
@@ -2,7 +2,10 @@
 if(NOT minizip_FOUND)
 	check_init_submodule(${PROJECT_SOURCE_DIR}/thirdparty/minizip)
 
+	set(MZ_ZLIB ON CACHE BOOL "Enable ZLIB compression, needed for DEFLATE")
+	set(MZ_BZIP2 OFF CACHE BOOL "Disable BZIP2 compression")
```
[the remark this answers is not in the document]
Comment (Jan200101) [reply cmake/Findminizip.cmake:6]: thats weird. It worked for me... I'll try to figure this out.

[the remark this answers is not in the document]
Comment (Alystrasz) [reply cmake/Findminizip.cmake:6]: Yep, same result after removing `build` and `game` directories.

(disclaimer: I don't have access to my Windows computer atm, so I'm using my Ubuntu working station with Docker building instructions)

[the remark this answers is not in the document]
Comment (Alystrasz) [reply cmake/Findminizip.cmake:6]: Reopening the project seemed to have some effect:

```text
[cmake] -- The following features have been enabled:
[cmake] 
[cmake]  * MZ_COMPAT, Enables compatibility layer
[cmake]  * MZ_ZLIB, Enables ZLIB compression
[cmake]  * MZ_PKCRYPT, Enables PKWARE traditional encryption
[cmake]  * MZ_ICONV, Enables iconv string encoding conversion library
[cmake] 
[cmake] -- The following features have been disabled:
[cmake] 
[cmake]  * MZ_BZIP2, Enables BZIP2 compression
[cmake]  * MZ_LZMA, Enables LZMA & XZ compression
[cmake]  * MZ_ZSTD, Enables ZSTD compression
[cmake]  * MZ_LIBCOMP, Enables Apple compression
[cmake]  * MZ_FETCH_LIBS, Enables fetching third-party libraries if not found
[cmake]  * MZ_FORCE_FETCH_LIBS, Enables fetching third-party libraries always
[cmake]  * MZ_WZAES, Enables WinZIP AES encryption
[cmake]  * MZ_OPENSSL, Enables OpenSSL for encryption
[cmake]  * MZ_LIBBSD, Builds with libbsd crypto random
[cmake]  * MZ_COMPRESS_ONLY, Only support compression
[cmake]  * MZ_DECOMPRESS_ONLY, Only support decompression
[cmake]  * MZ_FILE32_API, Builds using posix 32-bit file api
[cmake]  * MZ_BUILD_TESTS, Builds minizip test executable
[cmake]  * MZ_BUILD_UNIT_TESTS, Builds minizip unit test project
[cmake]  * MZ_BUILD_FUZZ_TESTS, Builds minizip fuzzer executables
[cmake]  * MZ_CODE_COVERAGE, Builds with code coverage flags
[cmake] 
[cmake] -- Configuring done (28.0s)
[cmake] -- Generating done (0.3s)
[cmake] -- Build files have been written to: /home/remy/Travail/NorthstarLauncher/build
```

[the remark this answers is not in the document]
Comment (Alystrasz) [reply cmake/Findminizip.cmake:6]: Even after removing both directories and all Docker images, and rebuilding from scratch, Docker seems to not give a fuck about the settings:

```text
-- The following features have been enabled:

 * MZ_COMPAT, Enables compatibility layer
 * MZ_ZLIB, Enables ZLIB compression
 * MZ_BZIP2, Enables BZIP2 compression
 * MZ_ZSTD, Enables ZSTD compression
 * MZ_FETCH_LIBS, Enables fetching third-party libraries if not found
 * MZ_PKCRYPT, Enables PKWARE traditional encryption
 * MZ_WZAES, Enables WinZIP AES encryption

-- The following features have been disabled:

 * MZ_LZMA, Enables LZMA & XZ compression
 * MZ_LIBCOMP, Enables Apple compression
 * MZ_FORCE_FETCH_LIBS, Enables fetching third-party libraries always
 * MZ_OPENSSL, Enables OpenSSL for encryption
 * MZ_LIBBSD, Builds with libbsd crypto random
 * MZ_ICONV, Enables iconv string encoding conversion library
 * MZ_COMPRESS_ONLY, Only support compression
 * MZ_DECOMPRESS_ONLY, Only support decompression
 * MZ_FILE32_API, Builds using posix 32-bit file api
 * MZ_BUILD_TESTS, Builds minizip test executable
 * MZ_BUILD_UNIT_TESTS, Builds minizip unit test project
 * MZ_BUILD_FUZZ_TESTS, Builds minizip fuzzer executables
 * MZ_CODE_COVERAGE, Builds with code coverage flags
 ```

**Review by Alystrasz — CHANGES_REQUESTED** [review]
I confirm disabling those methods does not have any negative impact on mod downloading (tested with https://github.com/R2Northstar/NorthstarLauncher/pull/595 branch).

---

Given we agreed on Discord to disable as many flags as possible, I tried disabling more, while ensuring mod downloading still works; I ended up with the following configuration:

```cmake
if(NOT minizip_FOUND)
	check_init_submodule(${PROJECT_SOURCE_DIR}/thirdparty/minizip)

	set(MZ_ZLIB ON CACHE BOOL "Disable ZLIB compression")
	set(MZ_BZIP2 OFF CACHE BOOL "Disable BZIP2 compression")
	set(MZ_LZMA OFF CACHE BOOL "Disable LZMA & XZ compression")
	set(MZ_PKCRYPT OFF CACHE BOOL "Disable PKWARE traditional encryption")
	set(MZ_WZAES OFF CACHE BOOL "Disable WinZIP AES encryption")
	set(MZ_ZSTD OFF CACHE BOOL "Disable ZSTD compression")
	set(MZ_SIGNING OFF CACHE BOOL "Disable zip signing support")

	add_subdirectory(${PROJECT_SOURCE_DIR}/thirdparty/minizip minizip)
	set(minizip_FOUND 1 PARENT_SCOPE)
endif()
```

**Comment by GeckoEidechse** [comment]
Given that minizip is only used for the not officially release mod-auto-download, I think we can be a lot more lax with review requirements on this PR \^\^

**Comment by Jan200101** [comment]
commited, rebased and merged (because why not)
Can be merged and released when ready

**Comment by GeckoEidechse** [comment]
Merging based on reviews

**Pull request merged**